### PR TITLE
WIP: Added FreeBSD SourceHut Builds

### PR DESCRIPTION
**Pull request by caughtquick**
Signed-off-by: caughtquick <me@caughtquick.tech>

Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [ ] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

### CODE REVIEW

File: .builds/freebsd.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+image: freebsd
```
Comment (Semisol) [inline .builds/freebsd.yml]: This is not a valid image?
Try one of these:
<details>
<summary>Suggestions</summary>

```suggestion
image: freebsd/latest
```
```suggestion
image: freebsd/13.x
```
```suggestion
image: freebsd/12.x
```
```suggestion
image: freebsd/11.x
```

</details>

Comment (caughtquick) [reply .builds/freebsd.yml]: Yep, I realized, going with 12.x

**Comment by Semisol** [comment]
> pkg: No packages available to install matching 'py36-pip' have been found in the repositories

You might want to check the package names.
(the arch one DID NOT fail, I just aborted it)

**Comment by Semisol** [comment]
@fossdd you screwed up the signoff, it has to be `name <email>` not `name email`

**Comment by fossdd** [comment]
> @fossdd you screwed up the signoff, it has to be `name <email>` not `name email`

Yes, I will force-push but my local copy is building so I do it in some minutes.

**Comment by nbsp** [comment]
Heads up that sourcehut only spins up three images for each build. If we want to go `archlinux`, `alpine`, `freebsd`, `debian` and potentially more BSD flavours, it'd pick three randomly and test them.

File: .builds/freebsd.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - wx28-gtk2-common 
+  - py38-pip
+  - gmake
+  - conan
+  - gcc
+sources:
+  - https://git.sr.ht/~tenacity/tenacity
+tasks:
+  - conan: |
+      sudo pip install conan
```
Comment (nbsp) [inline .builds/freebsd.yml]: You might want to build this branch against #228, and save the hassle of removing Conan once it's merged.

**Comment by caughtquick** [comment]
> Heads up that sourcehut only spins up three images for each build. If we want to go `archlinux`, `alpine`, `freebsd`, `debian` and potentially more BSD flavours, it'd pick three randomly and test them.

I think we should go with FreeBSD, Arch, and Alpine. It gives a mix or Glibc and Musl, stable and rolling and BSD and Linux.

**Comment by caughtquick** [comment]
@fossdd so we are install Conan through the FreeBSD package manager? If so then we should remove it installing through pip right?

**Comment by fossdd** [comment]
> @fossdd so we are install Conan through the FreeBSD package manager? If so then we should remove it installing through pip right?

Yes, the error you had is fixed by installing via FeeBSD package manager.

> > Heads up that sourcehut only spins up three images for each build. If we want to go `archlinux`, `alpine`, `freebsd`, `debian` and potentially more BSD flavours, it'd pick three randomly and test them.
> 
> I think we should go with FreeBSD, Arch, and Alpine. It gives a mix or Glibc and Musl, stable and rolling and BSD and Linux.

Agree.

**Comment by caughtquick** [comment]
@fossdd the changes didn't seem to fix the issue, I'm still getting the same error.

**Comment by nbsp** [comment]
[Here is a build that managed to get past the package manager](https://builds.sr.ht/~sfr/job/554089#bottom). It still fails, and the problem seems to be glibc.

Audacity is in FreeBSD's repos, you can try replicating the [package build](https://cgit.freebsd.org/ports/tree/audio/audacity).

**Comment by caughtquick** [comment]
@SFR-git has picked this up over on https://builds.sr.ht/~sfr.

**Comment by caughtquick** [comment]
@SFR-git has picked this up over on https://sr.ht/~sfr.

**Pull request closed without merging**